samples/snippets/src/main/java/events/ImportUserEventsInline.java

```diff
@@ -0,0 +1,198 @@
+  private static final String DEFAULT_CATALOG = String.format(
+      "projects/%s/locations/global/catalogs/default_catalog",
+      PROJECT_NUMBER);
+
+  private ImportUserEventsInline() {
+  }
+
+  /**
+   * Get user event service client.
+   *
+   * @return UserEventServiceClient.
+   * @throws IOException if endpoint is incorrect.
+   */
+  private static UserEventServiceClient getUserEventsServiceClient()
+      throws IOException {
+    return UserEventServiceClient.create();
+  }
+
+  /**
+   * Get user events for import.
+   *
+   * @return List<UserEvent>.
+   */
+  public static List<UserEvent> getUserEvents() {
+    List<UserEvent> userEvents = new ArrayList<>();
+
+    final int userEventsNumber = 3;
+
+    for (int i = 0; i < userEventsNumber; i++) {
+      Instant time = Instant.now();
+
+      Timestamp timestamp = Timestamp.newBuilder()
+          .setSeconds(time.getEpochSecond())
+          .build();
+
+      UserEvent userEvent = UserEvent.newBuilder()
+          .setEventType("home-page-view")
+          .setVisitorId("test_visitor_id")
+          .setEventTime(timestamp)
+          .build();
+
+      userEvents.add(userEvent);
+
+      System.out.printf("User Event: %s%n", i);
+      System.out.println(userEvent);
+    }
+
+    return userEvents;
+  }
+
+  /**
+   * Get import user events from inline source request.
+   *
+   * @param userEventsToImport list of user events to import.
+   * @return ImportUserEventsRequest.
+   */
+  public static ImportUserEventsRequest getImportEventsInlineSourceRequest(
+      final List<UserEvent> userEventsToImport) {
+    UserEventInlineSource inlineSource = UserEventInlineSource.newBuilder()
+        .addAllUserEvents(userEventsToImport)
+        .build();
+
+    UserEventInputConfig inputConfig = UserEventInputConfig.newBuilder()
+        .setUserEventInlineSource(inlineSource)
+        .build();
+
+    ImportUserEventsRequest importRequest = ImportUserEventsRequest.newBuilder()
+        .setParent(DEFAULT_CATALOG)
+        .setInputConfig(inputConfig)
+        .build();
+
+    System.out.printf("Import user events from inline source request: %s%n",
+        importRequest);
+
+    return importRequest;
+  }
+
+  /**
+   * Call the Retail API to import user events.
+   *
+   * @throws IOException          from the called method.
+   * @throws InterruptedException when a thread is waiting, sleeping, or
+   *                              otherwise occupied, and the thread is
+   *                              interrupted, either before or during the
+   *                              activity.
+   */
+  public static void importUserEventsFromInlineSource()
+      throws IOException, InterruptedException {
+
+    ImportUserEventsRequest importInlineRequest =
+        getImportEventsInlineSourceRequest(
+            getUserEvents());
+
+    UserEventServiceClient serviceClient = getUserEventsServiceClient();
+
+    String operationName = serviceClient
+        .importUserEventsCallable()
```

Comment:
Inline import for user events/products are generally fast, we may use the old style of using
serviceClient.importUserEventsAsync(importInlineRequest) to return the OperationFuture.

Reply:
done